### ISSUE TITLE
pacman example

### ISSUE THREAD

**Issue by Wang-Yue** [ageron/handson-ml#63]
I have been running the example for weeks. currently it just reached 100000000 iterations but pacman still behaves very naively. 

I guess something is wrong. 

Each game has 3 lives. The code may give poor weights for the actions taken right before pacman dies, but it may give good weights for the actions right before first two kills. I suspect that's the reason pacman never learn to behave properly.

**Comment by ageron** [comment]
Thanks for your feedback.  That's odd, maybe I introduced a bug at some point.  I'll look into this when I get back from vacation next week.  Can you try with a simpler Atari game, such as Pong?

**Comment by Wang-Yue** [comment]
My code is listed in https://gist.github.com/Wang-Yue/baf7bfe348d7ca8a31c71e613ae9a5ee . It's mostly just copy and paste from your notebook. 

BTW, I just finished reading your book, it's the best deep learning book I've ever read, and I learnt a lot. Thank you so much!

**Comment by Pibben** [comment]
I have the same issue. It improves for the first 50.000 steps or so. But I ran it for another 500.000 steps without further improvements.

**Comment by ageron** [comment]
Okay, I found the issues, it was pretty bad. In fact, the worst errors in the book. :(( You will find the details in the notebook for chapter 16, I just updated it.  Now that the issues are fixed, in just a few hours (with a good GPU card) Ms Pacman clearly starts to make good progress.
I'm really sorry about these issues, I guess I was a bit too eager to finish the book, it was the last section of the last chapter, so I did not check as thoroughly as I did in the rest of book.
